src/Uno.Toolkit.UI/Controls/ResponsiveView/ResponsiveView.cs

```diff
@@ -22,26 +22,33 @@ public ResponsiveView()
 		DefaultStyleKey = typeof(ResponsiveView);
 
 		Loaded += OnLoaded;
+		Unloaded += OnUnloaded;
 	}
 
+
 	private void OnLoaded(object sender, RoutedEventArgs e)
 	{
 		if (XamlRoot is null) return;
 
+		ResponsiveHelper.InitializeIfNeeded(XamlRoot);
+		ResponsiveHelper.WindowSizeChanged += OnWindowSizeChanged;
+
 		UpdateTemplate(forceApplyValue: true);
+	}
 
-		XamlRoot.Changed -= OnXamlRootPropertyChanged;
-		XamlRoot.Changed += OnXamlRootPropertyChanged;
+	private void OnUnloaded(object sender, RoutedEventArgs e)
+	{
+		ResponsiveHelper.WindowSizeChanged -= OnWindowSizeChanged;
 	}
 
-	private void OnXamlRootPropertyChanged(XamlRoot sender, XamlRootChangedEventArgs args)
+	private void OnWindowSizeChanged(object sender, Size size)
 	{
-		if (sender.Size == LastResolved?.Size) return;
+		if (size == LastResolved?.Size) return;
 
 		UpdateTemplate();
 	}
 
-	internal void ForceResponsiveSize(Size size)
+	public void ForceResponsiveSize(Size size)
```

Comment:
Should we add `[EditorBrowsable(EditorBrowsableState.Never)]` here?

Reply:
mroe like it shouldnt be made public, since there are better alternatives now